##Disciplina - Introdução a Ciência de Dados - Univesp
##Exercícios de Apoio da Semana 2
##**GABARITO**

###1 - Carregue as bibliotecas NumPy e Pandas

In [2]:
import numpy as np
import pandas as pd

###2 - Crie um NDArray com os seguintes valores [1, 2.5, 7, -78, -0.9, 130, 25]

In [3]:
lista=[1, 2.5, 7, -78, -0.9, 130, 25]

In [4]:
dados=np.array(lista)

In [5]:
dados

array([  1. ,   2.5,   7. , -78. ,  -0.9, 130. ,  25. ])

###3 - Crie um NDArray com os valores de 0 a 10

In [6]:
dados=np.arange(11)

In [7]:
dados

array([ 0,  1,  2,  3,  4,  5,  6,  7,  8,  9, 10])

###4 – Crie um NDArray de dimensão 2x5 com valores 1

In [10]:
dados=np.ones((2,5))

In [11]:
dados

array([[1., 1., 1., 1., 1.],
       [1., 1., 1., 1., 1.]])

###5 – Crie um NDArray com valores de 1 a 100, com intervalos de 3 em 3.

In [12]:
dados=np.arange(1, 100, 3)

In [13]:
dados

array([ 1,  4,  7, 10, 13, 16, 19, 22, 25, 28, 31, 34, 37, 40, 43, 46, 49,
       52, 55, 58, 61, 64, 67, 70, 73, 76, 79, 82, 85, 88, 91, 94, 97])

###6 – Mostre as raízes quadradas dos valores do NDArray criado no exercícios 5.

In [18]:
np.sqrt(dados)

array([1.        , 2.        , 2.64575131, 3.16227766, 3.60555128,
       4.        , 4.35889894, 4.69041576, 5.        , 5.29150262,
       5.56776436, 5.83095189, 6.08276253, 6.32455532, 6.55743852,
       6.78232998, 7.        , 7.21110255, 7.41619849, 7.61577311,
       7.81024968, 8.        , 8.18535277, 8.36660027, 8.54400375,
       8.71779789, 8.88819442, 9.05538514, 9.21954446, 9.38083152,
       9.53939201, 9.69535971, 9.8488578 ])

###7 – Altere a décima posição do NDArray criado no exercício 5 com o dobro do seu valor.

In [19]:
dados[9]=dados[9]*2

In [20]:
dados

array([ 1,  4,  7, 10, 13, 16, 19, 22, 25, 56, 31, 34, 37, 40, 43, 46, 49,
       52, 55, 58, 61, 64, 67, 70, 73, 76, 79, 82, 85, 88, 91, 94, 97])

####Obs: veja que a posição 9 tinha o valor 28 e passou a ter 56

###8 – Execute o comando para criar um Dataframe com os dados disponíveis no notebook.

In [22]:
dados={'Empresa':['Microsoft', 'Google','Univesp','NVData Analytics','Power Bi Experience','USP'], 'Qtde Func':[1000, 1500,500, 100,100,499], 'Site':['https://www.microsoft.com/pt-br','https://www.google.com/','https://univesp.br/','https://www.nvdata.com.br/','https://powerbiexperience.com/pt/','https://www.usp.br/'],'Instagram':['@microsoft','@google','@univespoficial','@nv_data','@leokarpa','@usp.oficial'],'Criação':[1975,1998,2012,2019,2016,1934], 'País':['EUA','EUA','Brasil','Brasil','Brasil','Brasil']}

In [24]:
df1=pd.DataFrame(dados)

In [25]:
df1

Empresa  Qtde Func  ... Criação    País
0            Microsoft       1000  ...    1975     EUA
1               Google       1500  ...    1998     EUA
2              Univesp        500  ...    2012  Brasil
3     NVData Analytics        100  ...    2019  Brasil
4  Power Bi Experience        100  ...    2016  Brasil
5                  USP        499  ...    1934  Brasil

[6 rows x 6 columns]

###9 – Execute um comando que seja possível identificar qual a média da quantidade de funcionários que as empresas listadas tem. Informe também a soma dos funcionários que as empresas tem.

In [28]:
df1.describe()

Qtde Func      Criação
count     6.000000     6.000000
mean    616.500000  1992.333333
std     545.631286    32.843061
min     100.000000  1934.000000
25%     199.750000  1980.750000
50%     499.500000  2005.000000
75%     875.000000  2015.000000
max    1500.000000  2019.000000

####Obs: veja que a média dos funcionarios de todas as empresas é 616.5 (observe o mean)

In [29]:
df1['Qtde Func'].sum()

3699

###10 – Mostre as primeiras 3 linhas do dataframe.

In [30]:
df1.head(3)

Empresa  Qtde Func  ... Criação    País
0  Microsoft       1000  ...    1975     EUA
1     Google       1500  ...    1998     EUA
2    Univesp        500  ...    2012  Brasil

[3 rows x 6 columns]

###11 – Mostre as últimas 3 linhas do dataframe.

In [31]:
df1.tail(3)

Empresa  Qtde Func  ... Criação    País
3     NVData Analytics        100  ...    2019  Brasil
4  Power Bi Experience        100  ...    2016  Brasil
5                  USP        499  ...    1934  Brasil

[3 rows x 6 columns]

###12 – Liste apenas as linhas em que as empresas foram criadas nos últimos 10 anos. Informe a quantidade de registros.

In [33]:
df1[df1['Criação']>=2011]

Empresa  Qtde Func  ... Criação    País
2              Univesp        500  ...    2012  Brasil
3     NVData Analytics        100  ...    2019  Brasil
4  Power Bi Experience        100  ...    2016  Brasil

[3 rows x 6 columns]

###13 – Vamos imaginar que em todas as empresas exista 1 chefe para cada 23 funcionários. Crie uma coluna que mostre a quantidade de chefes que cada empresa teria e depois informe quantos chefes teríamos ao todo. (os valores não precisam ser números inteiros, é apenas uma estimativa).

In [34]:
df1['Chefes']=df1['Qtde Func']/23

In [35]:
df1

Empresa  Qtde Func  ...    País     Chefes
0            Microsoft       1000  ...     EUA  43.478261
1               Google       1500  ...     EUA  65.217391
2              Univesp        500  ...  Brasil  21.739130
3     NVData Analytics        100  ...  Brasil   4.347826
4  Power Bi Experience        100  ...  Brasil   4.347826
5                  USP        499  ...  Brasil  21.695652

[6 rows x 7 columns]

In [36]:
df1.Chefes.sum()

160.82608695652175